### PR TITLE
Refactored ServerConfig to serialize and deserialize the same instance

### PR DESCRIPTION
**Pull request by Measurity**
- [x] Verify all cases where config serialization happens are handled.

### CODE REVIEW

File: NitroxServer-Subnautica/Program.cs [hunk 1]
```diff
@@ -262,7 +269,6 @@ private static void StopAndExitServer()
         {
             Log.Info("Exiting ...");
             Server.Instance.Stop();
-            Environment.Exit(0);
```
Comment (Jannify) [inline NitroxServer-Subnautica/Program.cs:265]: Wouldn't this cause the cmd window (if you choose this option) to stay and not automatically close?

File: NitroxServer-Subnautica/Program.cs [hunk 2]
```diff
@@ -32,7 +33,7 @@ public class Program
         private static async Task Main(string[] args)
         {
             // The thread that writers to console is paused while selecting text in console. So console writer needs to be async.
-            Log.Setup(asyncConsoleWriter: true, isConsoleApp: true);
+            Log.Setup(true, isConsoleApp: true);
```
Comment (Jannify) [inline NitroxServer-Subnautica/Program.cs:36]: Why removing just one side? All or nothing :P

Comment (Measurity) [reply NitroxServer-Subnautica/Program.cs:36]: It formatted this automatically.

File: NitroxServer-Subnautica/Program.cs (hunk 1 above)
Comment (Measurity) [reply NitroxServer-Subnautica/Program.cs:265]: It still automatically closes when clicking exit button on console window or CTRL+C. Though naming could be changed, will do.

**Pull request merged**